fe/fe-core/src/main/java/com/starrocks/external/HiveMetaStoreTableUtils.java

```diff
@@ -216,23 +223,112 @@ public static Type convertColumnType(String hiveType) throws DdlException {
         }
     }
 
+    // this func targets at convert hudi column type(avroSchema) to starrocks column type(primitiveType)
+    public static Type convertHudiTableColumnType(Schema avroSchema) throws DdlException {
+        Schema.Type columnType = avroSchema.getType();
+        LogicalType logicalType = avroSchema.getLogicalType();
+        PrimitiveType primitiveType = null;
+        boolean isString = false;
+        boolean isConvertedFailed = false;
+
+        switch (columnType) {
+            case BOOLEAN:
+                primitiveType = PrimitiveType.BOOLEAN;
+                break;
+            case INT:
+                if (logicalType instanceof LogicalTypes.Date) {
+                    primitiveType = PrimitiveType.DATE;
+                } else if (logicalType instanceof LogicalTypes.TimeMillis) {
+                    primitiveType = PrimitiveType.TIME;
+                } else {
+                    primitiveType = PrimitiveType.INT;
+                }
+                break;
+            case LONG:
+                if (logicalType instanceof LogicalTypes.TimeMicros) {
+                    primitiveType = PrimitiveType.TIME;
+                } else if (logicalType instanceof LogicalTypes.TimestampMillis
+                        || logicalType instanceof LogicalTypes.TimestampMicros) {
+                    primitiveType = PrimitiveType.DATETIME;
+                } else {
+                    primitiveType = PrimitiveType.BIGINT;
+                }
+                break;
+            case FLOAT:
+                primitiveType = PrimitiveType.FLOAT;
+                break;
+            case DOUBLE:
+                primitiveType = PrimitiveType.DOUBLE;
+                break;
+            case STRING:
+                isString = true;
+                break;
+            case ARRAY:
+                Type type = Utils.convertToArrayType(avroSchema);
+                if (type.isArrayType()) {
+                    return type;
+                }
```

Comment:
what if this condition is false?

Reply:
> what if this condition is false?

```
case ARRAY:
    Type type = Utils.convertToArrayType(avroSchema);
    if (type.isArrayType()) {
        return type;
    } else {
        isConvertedFailed = false;
        break;
    }
...
default:
    isConvertedFailed = true;
    break;
    }

if (isConvertedFailed) {
    throw new DdlException(String.format(COLUMN_CONVERTED_FAILED_MSG, avroSchema.getType()));
}

return ScalarType.createType(primitiveType);
```
If `type.isArrayType()` is false, referring to `convertHiveTableColumnType`, type validate should fail here.